edison/static/js/policy.controller.js

```diff
@@ -148,8 +146,11 @@ function addPolicy() {
 
 function saveCondition() {
     addCondition(policy);
+    var countCondition = policy.getCountCondition();
     showCondition(policy, countCondition, 'condition-list');
-    countCondition++;
+    var count = policy.getCountCondition();
+    count++;
+    policy.setCountCondtion(count);
 }
```

Comment:
I think it will be better to increase the countCondition in the addCondition function inside Policy class.
you wanted this variable to be private and right now you can change its values from outside.
i see why you did it, you want the use of the correct place in the array in the showCondition function.
so what i this you should do is:
1. add the counter++ at the end of addCondition function
2. remove her the count++, also remove the setCountCondition function from policy
3. whenever you need to use this variable location use [countCondition-1]

Reply:
Ok :)